ios/brave-ios/Sources/BraveShared/Extensions/URLExtensions.swift

```diff
@@ -216,6 +216,96 @@ extension URL {
     return components.url ?? self
   }
 
+  /// Returns the base domain from a given hostname unless URL is IPV6 or does not
+  /// contain a `.`. The base domain name is defined as the public domain suffix
+  /// with the base private domain attached to the front. For example, for the
+  /// URL www.bbc.co.uk, the base domain would be bbc.co.uk. The base domain
+  /// includes the public suffix (co.uk) + one level down (bbc).
+  /// - parameter fallbackToHost: if it fallsback to host
+  ///   if the registry is empty
+  /// - returns The base domain string for the given host name.
+  public func baseDomain(fallbackToHost: Bool) -> String? {
+    guard !isIPv6, let host = host else { return nil }
+
+    // If this is just a hostname and not a FQDN, use the entire hostname.
+    if !host.contains(".") {
+      return host
+    }
+
+    let registry = (self as NSURL).domainAndRegistry
+    if !registry.isEmpty {
+      return registry
+    }
+
+    if fallbackToHost {
```

Comment:
I think we may not need it anymore as fallbacking to host should be safe

Reply:
I've removed the fallback option in [3f552d3](https://github.com/brave/brave-core/pull/28730/commits/3f552d32b283b859ef77522d181abb3662b825a3) 👍